comid-triples.cddl

```diff
@@ -0,0 +1,172 @@
+concise-mid-tag = {
+  ; TODO(tho) language
+  ; TODO(tho) tag-id
+  ; TODO(tho) module-name
+  ; TODO(tho) entity
+  ; TODO(tho) linked-tags
+  comid.triples => triples-map
+  ; TODO(tho) extension
+}
+
+triples-map = non-empty<{
+  ? comid.reference-triples => one-or-more<reference-triple-record>
+  ? comid.endorsed-triples => one-or-more<endorsed-triple-record>
+  ? comid.attest-key-triples => one-or-more<attest-key-triple-record>
+  ? comid.identity-triples => one-or-more<identity-triple-record>
+  * $$triples-map-extension
+}>
+
+; REFERENCE-VALUE triple
+; "target environment $TE" "has reference measurements" "$RV"
+; NOTE(tho): can an attesting evironement have reference values?
+reference-triple-record = [
+  environment-map ; target environment
+  one-or-more<measurement-map> ; reference measurements
+]
+
+; ENDORSED-VALUE triple
+; "environment $E" "has endorsed measurements" "$EV"
+endorsed-triple-record = [
+  environment-map ; (target or attesting) environment
+  one-or-more<measurement-map> ; endorsed measurements
+]
+
+; ATTESTATION-VERIFICATION-KEY triple
+; "attesting environment $AE" "signs Evidence that can be verified with key" "$K"
+attest-key-triple-record = [
+  environment-map ; attesting environment
+  one-or-more<verification-key> ; attestation verification key(s)
+]
+
+; DEVICE-IDENTITY triple
+; "device $D" "is identified by key" "$K"
+identity-triple-record = [
+  environment-map ; device identifier (instance or class)
+  one-or-more<verification-key> ; DevID, or semantically equivalent
+]
+
+verification-key = {
+  ; Verification key (typically a public key)
+  comid.key => COSE_Key
+  ; Optional X.509 certificate chain corresponding to the public key
+  ; in comid.key, encoded as an array of one or more base64-encoded
+  ; DER PKIX certificates.  The certificate containing the public key
+  ; in comid.key MUST be the first certificate.  This MAY be followed
+  ; by additional ; certificates, with each subsequent certificate
+  ; being the one used to certify the previous one.
+  ? comid.keychain => [ + tstr ]
+}
+
+environment-map = non-empty<{
+  ? comid.class => class-map
+  ? comid.instance => $instance-id-type-choice
+}>
+
+class-map = {
+  comid.class-id => $class-id-type-choice
+  ? comid.vendor => tstr
+  ? comid.model => tstr
+  ? comid.layer => uint
+  ? comid.index => uint
+}
+
+; generic name-spaced identifier
+ns-auth = tstr
+ns-id<id> = [ ns-auth, id ]
```

Comment:
is <id> a todo item or CDDL-ism  / CBOR-ism?

Reply:
cddl-ism (see https://www.rfc-editor.org/rfc/rfc8610.html#section-3.10)